lasagne/layers/shape.py

```diff
@@ -88,6 +89,11 @@ def __init__(self, incoming, shape, **kwargs):
                 if len(s) != 1 or not isinstance(s[0], int) or s[0] < 0:
                     raise ValueError("`shape` input references must be "
                                      "single-element lists of int >= 0")
+            elif isinstance(s, T.TensorVariable):
+                if s.ndim != 0:
+                    raise ValueError(
+                        "The symbolic variable specifying shape must be a "
```

Comment:
Maybe "A symbolic variable in a shape specification must be a scalar, but had %i dmensions"? The variable technically doesn't specify the shape, it only specifies one dimension.